### ISSUE TITLE
public key for recursive source code

### ISSUE THREAD

**Issue by ghost** [AdaptiveCode/AdaptiveCode#2]
Do you need a public key to download the recursive source code?

**Comment by ghost** [comment]
Here is an image of my bash session:
![image](https://cloud.githubusercontent.com/assets/25227667/25961266/8db42d7a-362e-11e7-8b6f-45fa7bd179c6.png)

**Comment by neman** [comment]
There must be some issue with your machine. I cloned this repo with http and ssl successfully.

**Comment by wapxmas** [comment]
The same issue happens on my machine (Windows 7 x64).

```
> git clone --recursive git@github.com:AdaptiveCode/AdaptiveCode.git
Cloning into 'AdaptiveCode'...
Permission denied (publickey).
fatal: Could not read from remote repository.
```

Although I have managed to download these chapters one-by-one, that is:
```
git clone https://github.com/AdaptiveCode/Chapter1
git clone https://github.com/AdaptiveCode/Chapter3
git clone https://github.com/AdaptiveCode/Chapter4
git clone https://github.com/AdaptiveCode/Chapter5and6
git clone https://github.com/AdaptiveCode/Chapter7
git clone https://github.com/AdaptiveCode/Chapter9
git clone https://github.com/AdaptiveCode/Chapter10
```

**Comment by garymcleanhall** [comment]
I haven't merged in the `2e` branch yet, anyway, so you're probably better off cloning each chapter individually for now.

You probably don't have ssh configured correctly, which is why cloning over https works. 

Try this instead:

```
> git clone --recursive https://github.com/AdaptiveCode/AdaptiveCode.git
```

**Comment by wapxmas** [comment]
Guidelines I have used from this [answer](https://stackoverflow.com/a/27176952/2526001) helped me to configure ssh correctly, so now I can clone repo with
```
git clone --recursive git@github.com:AdaptiveCode/AdaptiveCode.git
```
Although it`s empty.

**Comment by garymcleanhall** [comment]
Yeah, `master` is empty until I finish porting/writing the code for the 2nd Edition.

Git's `clone` command accepts a `--branch` parameter, and the branch with all the 2nd Edition code on is called `2e`. So, you could do this:

```
git clone --branch 2e --recursive git@github.com:AdaptiveCode/AdaptiveCode.git
```

**Comment by hjr2000** [comment]
For a £30/$50 book I find the fact that the code repo is still not ready in later 2017 to be pretty poor. And the fact that the community have had to fill in the blanks with regards to how to clone stuff! Not really good enough, sorry Gary.

**Comment by wapxmas** [comment]
To be honest, I think it is not the matter of this book to bear any code repo thus I am not disappointed by the fact that code repo is not ready now, but I agree if something like repo is mentioned in the book it must be ready for use.

**Comment by garymcleanhall** [comment]
@hjr2000 Hi Haden, apologies you are dissatisfied, but there is a lot of code already there from the 1st edition that is still very relevant. The book was not rewritten, nor did it need to be, it was just updated to make corrections and improvements and add depth where necessary. Anyway, hope you got something from the book, or you were successful in getting a refund 👍

**Comment by EpsilonSigma** [comment]
Hello, how do I get a refund and from whom? Last time I contacted Microsoft Press about a similar situation (Fritz's book on ASP.NET Core), they did not remedy the situation. I won't have another book which I can't use. I pay money for these, you know, I don't get them off free download sites.

**Comment by garymcleanhall** [comment]
@EpsilonSigma How can't you "use" the book, out of interest? It stands alone as a reference and there are a lot of samples here. Curious as to what is missing so that I might provide it to you.

**Comment by EpsilonSigma** [comment]
The publisher asked you to deliver both a book and accompanying code. You have yet not delivered the code. I cloned the repository following the instructions in the book and there was nothing there. I have to jump through hoops in order to clone the incomplete branch you’ve created for the second edition.

Just deliver what you were supposed to, Gary, instead of attacking your readers.

In any case, I’ve received a refund and will not read your book. I only buy quality books, which have been written, edited and published with care.

Have a nice day.

**Comment by garymcleanhall** [comment]
Err, I didn't attack you at all! Quite an overreaction.

Shame you have decided not to read the book - which you might still learn something from? - because of a misunderstanding around the code samples. 

I'll defend myself here because the book is clearly of good quality - there wouldn't be a 2nd edition if the 1st edition wasn't received so well. And the 2nd edition is an *improvement* on the first.

Anyway, glad you got a refund and hope you reconsider cos there are lots of useful things in the book for all sorts of developers.

Thanks,
Gary
